File: app/src/main/java/com/harvey/image/CustomCachingGlideModule.java
package com.harvey.image;

import android.content.Context;

import com.bumptech.glide.Glide;
import com.bumptech.glide.GlideBuilder;
import com.bumptech.glide.load.DecodeFormat;
import com.bumptech.glide.load.engine.bitmap_recycle.LruBitmapPool;
import com.bumptech.glide.load.engine.cache.ExternalCacheDiskCacheFactory;
import com.bumptech.glide.load.engine.cache.LruResourceCache;
import com.bumptech.glide.load.engine.cache.MemorySizeCalculator;
import com.bumptech.glide.module.GlideModule;


/**
 * Created by harvey on 2016/9/1 0001 11:44
 */
public class CustomCachingGlideModule implements GlideModule {
	public static final String GLIDE_CACHE_PATH = "Avatar";
	@Override
	public void applyOptions(Context context, GlideBuilder builder) {
		builder.setDecodeFormat(DecodeFormat.DEFAULT);
		// 自定义内存缓存20%
		MemorySizeCalculator calculator = new MemorySizeCalculator(context);
		int defaultMemoryCacheSize = calculator.getMemoryCacheSize();
		int defaultBitmapPoolSize = calculator.getBitmapPoolSize();
		int customMemoryCacheSize = (int) (1.2 * defaultMemoryCacheSize);
		int customBitmapPoolSize = (int) (1.2 * defaultBitmapPoolSize);
		builder.setMemoryCache(new LruResourceCache(customMemoryCacheSize));
		builder.setBitmapPool(new LruBitmapPool(customBitmapPoolSize));
		int cacheSize100MegaBytes = 104857600;
		builder.setDiskCache(new ExternalCacheDiskCacheFactory(context, GLIDE_CACHE_PATH, cacheSize100MegaBytes));
	}

	@Override
	public void registerComponents(Context context, Glide glide) {

	}
}

File: kotlin-app/src/main/java/com/harvey/image/CustomCachingGlideModule.kt
package com.harvey.image

import android.content.Context
import com.bumptech.glide.Glide
import com.bumptech.glide.GlideBuilder
import com.bumptech.glide.load.DecodeFormat
import com.bumptech.glide.load.engine.bitmap_recycle.LruBitmapPool
import com.bumptech.glide.load.engine.cache.ExternalCacheDiskCacheFactory
import com.bumptech.glide.load.engine.cache.LruResourceCache
import com.bumptech.glide.load.engine.cache.MemorySizeCalculator
import com.bumptech.glide.module.GlideModule

/**
 * Created by hanhui on 2018/6/1 0001 11:49
 */
class CustomCachingGlideModule : GlideModule {

    override fun applyOptions(context: Context, builder: GlideBuilder) {
        builder.setDecodeFormat(DecodeFormat.DEFAULT)
        val calculator = MemorySizeCalculator(context)
        val defaultMemoryCacheSize = calculator.memoryCacheSize
        val defaultBitmapPoolSize = calculator.bitmapPoolSize
        val customMemoryCacheSize = (1.2 * defaultMemoryCacheSize).toInt()
        val customBitmapPoolSize = (1.2 * defaultBitmapPoolSize).toInt()
        builder.setMemoryCache(LruResourceCache(customMemoryCacheSize))
        builder.setBitmapPool(LruBitmapPool(customBitmapPoolSize))
        val cacheSize100MegaBytes = 104857600
        builder.setDiskCache(ExternalCacheDiskCacheFactory(context, GLIDE_CACHE_PATH, cacheSize100MegaBytes))
    }

    override fun registerComponents(context: Context, glide: Glide) {

    }

    companion object {
        val GLIDE_CACHE_PATH = "Avatar"
    }
}